src/overlays/actors/ovl_En_Zl1/z_en_zl1.c

```diff
@@ -15,51 +15,650 @@ void EnZl1_Destroy(Actor* thisx, GlobalContext* globalCtx);
+}*/
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4B240.s")
+#endif
+
+void func_80B4B7F4(CsCmdActorAction* actorAction, Vec3f* pos) {
+    pos->x = actorAction->startPos.x;
+    pos->y = actorAction->startPos.y;
+    pos->z = actorAction->startPos.z;
+}
+
+void func_80B4B834(CsCmdActorAction* actorAction, Vec3f* pos) {
+    pos->x = actorAction->endPos.x;
+    pos->y = actorAction->endPos.y;
+    pos->z = actorAction->endPos.z;
+}
+
+void func_80B4B874(EnZl1* this, GlobalContext* globalCtx) {
+    this->skelAnime.flags |= 1;
+    SkelAnime_LoadAnimationType5(globalCtx, &this->actor, &this->skelAnime, 1.0f);
+}
+
+void func_80B4B8B4(Actor* thisx, GlobalContext* globalCtx) {
+    AH_A spB0 = D_80B4E67C;
+    u8_A spA4 = D_80B4E6A4;
+    Vec3f sp98 = D_80B4E6B0;
+    Vec3f sp8C = D_80B4E6BC;
+    EnZl1* this = THIS;
+    f32 actionLength;
+    CsCmdActorAction* actorAction;
+    Vec3f sp74;
+    Vec3f sp68;
+    Vec3f velocity = D_80B4E6C8;
+    s32 pad;
+    f32 frameCount;
+    Vec3f sp48;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80B4B874(thisx, globalCtx);
+    if (globalCtx->csCtx.state == 0) {
+        this->actionFunc = func_80B4BBC4;
+        return;
+    }
+
+    actorAction = globalCtx->csCtx.npcActions[0];
+    if (actorAction != NULL) {
+        func_80B4B7F4(actorAction, &sp74);
+        func_80B4B834(actorAction, &sp68);
+        if (this->unk_1E6 == 0) {
+            sp48 = sp74;
+            thisx->initPosRot.pos = sp48;
+            thisx->posRot.pos = sp48;
+        }
+        if (this->unk_1E6 != actorAction->action) {
+            frameCount = SkelAnime_GetFrameCount(&spB0.unk_00[actorAction->action]->genericHeader);
+            SkelAnime_ChangeAnim(&this->skelAnime, spB0.unk_00[actorAction->action], 1.0f, 0.0f, frameCount,
+                                 spA4.unk_00[actorAction->action], -10.0f);
+            this->unk_1E6 = actorAction->action;
+        }
+        thisx->velocity = velocity;
+        if (globalCtx->csCtx.frames < actorAction->endFrame) {
+            actionLength = actorAction->endFrame - actorAction->startFrame;
+            thisx->velocity.x = (sp68.x - sp74.x) / actionLength;
+            thisx->velocity.y = (sp68.y - sp74.y) / actionLength;
+            thisx->velocity.y += thisx->gravity;
+            if (thisx->velocity.y < thisx->minVelocityY) {
+                thisx->velocity.y = thisx->minVelocityY;
+            }
+            thisx->velocity.z = (sp68.z - sp74.z) / actionLength;
+        }
+        func_80038290(globalCtx, thisx, &this->unk_200, &this->unk_206, thisx->posRot2.pos);
+        func_800C04D8(globalCtx, this->unk_1E8, &sp98, &sp8C);
+        func_800C0704(globalCtx, this->unk_1E8, 70.0f);
+    }
+}
+
+void func_80B4BBC4(Actor* thisx, GlobalContext* globalCtx) {
+    EnZl1* this = THIS;
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000438.genericHeader);
```

Comment:
it seems basically this whole file is taking thisx so i wont coment on it anymore, but i would change all of them except for the entry point functions